icu4j/demos/src/com/ibm/icu/dev/demo/rbnf/RbnfDemo.java

```diff
@@ -108,37 +112,40 @@ public void textValueChanged(TextEvent e) {
                 parsePosition.setIndex(0);
                 Number temp = numberFormatter.parse(fieldText, parsePosition);
                 if (temp == null || parsePosition.getIndex() == 0) {
-                    theNumber = 0;
+                    theNumber = BigDecimal.ZERO;
                     textField.setText("PARSE ERROR");
                 }
                 else {
-                    theNumber = temp.doubleValue();
-                    textField.setText(spelloutFormatter.format(theNumber, ruleSetName));
+                    theNumber = new BigDecimal(fieldText);
+                    textField.setText(spelloutFormatter.format(theNumber.doubleValue(), ruleSetName));
```

Comment:
I will admit that I initially did this quick hack when doing the demo for the [Number Format Tester](https://st.unicode.org/cldr-apps/numbers.jsp). If you're trying to do this accurately, then you should at least check if it has a fraction. If it doesn't have a fraction, then you should use the long value instead.

You may want to look at the more maintained Number Format Tester here: https://github.com/unicode-org/cldr/blob/master/tools/cldr-apps/WebContent/numbers.jsp